### PR TITLE
IP Spaces CRUD

### PR DESCRIPTION
**Pull request by Didainius**
This PR adds support for IP Spaces. It is the first PR which exposes CRUD API for IP Spaces. Next PRs will contain the integration part with the rest of components.

### CODE REVIEW

File: govcd/ip_space.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+		return nil, err
+	}
+
+	typeResponses := []*types.IpSpace{{}}
+	err = client.OpenApiGetAllItems(apiVersion, urlRef, queryParameters, &typeResponses, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wrap all typeResponses into DefinedEntityType types with client
```
Comment (adezxc) [inline govcd/ip_space.go:80]: This is probably leftover from other code

**Review by adezxc — APPROVED** [review]

File: types/v56/ip_space.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	// AllocatedCount holds the number of allocated IP addresses or IP Prefixes.
+	AllocatedCount string `json:"allocatedCount"`
+	// UsedCount holds the number of used IP addresses or IP Prefixes. An allocated IP address or IP
+	// Prefix is considered used if it is being used in network services such as NAT rule or in Org
+	// VDC network definition.
+	UsedCount string `json:"usedCount"`
+	// UnusedCount holds the number of unused IP addresses or IP Prefixes. An IP address or an IP
+	// Prefix is considered unused if it is allocated but not being used by any network service or
+	// any Org vDC network definition.
+	UnusedCount string `json:"unusedCount"`
```
[the remark this answers is not in the document]
Comment (lvirbalas) [reply types/v56/ip_space.go:91]: Interesting, same question could be for:

https://github.com/vmware/go-vcloud-director/pull/578/files#diff-91f329c83083b4e72429075d5f926a8b4dc4f47363c52e17c1f2b950a0edeeb1R162-R165

File: govcd/ip_space.go (hunk 1 above)
Comment (Didainius) [reply govcd/ip_space.go:80]: Fixed.

File: govcd/ip_space.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+		return nil, fmt.Errorf("error ")
+	}
+
+	return vcdClient.GetIpSpaceById(singleIpSpace.IpSpace.ID)
+}
+
+// GetIpSpaceByNameAndOrgId retrieves IP Space with a given name in a particular Org
+// Note. Only PRIVATE IP spaces belong to Orgs
+func (vcdClient *VCDClient) GetIpSpaceByNameAndOrgId(name, orgId string) (*IpSpace, error) {
+	if name == "" {
```
Comment (adambarreiro) [inline govcd/ip_space.go:149]: Just wondering whether we should also check `orgId == ""`

**Review by adambarreiro — APPROVED** [review]

**Review by dataclouder — APPROVED** [review]
Tests pass in all versions
LGTM

File: govcd/ip_space.go (hunk 3 above)
Comment (Didainius) [reply govcd/ip_space.go:149]: Added

**Pull request merged**